src/optics.jl

```diff
@@ -134,6 +135,10 @@ else
     using Base: Returns
 end
 
+struct Changed end
```

Comment:
How important is it, that we have `Changed/Unchanged`? It is a pure optimization and does not influence semantics, so we could drop it later at will if we wanted, right?

Reply:
Ok that makes a ton of sense.